### DISCUSSION TITLE
arm-uwp is being deprecated, arm64-uwp should be promoted to a non-Community triplet

### DISCUSSION THREAD

**Discussion by walbourn** [microsoft/vcpkg#33481] [Ideas]
The ARM32 platform is being actively deprecated. See [Microsoft Learn](https://learn.microsoft.com/en-us/windows/arm/arm32-to-arm64)

Currently only *arm-uwp* is a 'supported' triplet, and *arm64-uwp* is a community triplet. The *arm-uwp* triplet support should be retired, and *arm64-uwp* should be promoted to a non-community triplet.

**Comment by walbourn** [comment]
At a minimum, I believe *arm-uwp* needs to be moved to a community triplet and *arm64-uwp* needs to be made a fully supported triplet.

**Reply by autoantwort** [reply]
I have created https://github.com/microsoft/vcpkg/pull/33485 for that

**Comment by walbourn** [comment]
The CI build was swapped in #33759

**Comment by walbourn** [comment]
@BillyONeal 

In addition to swapping the CI build, I think we need to add to the ``arm-*`` triplets a deprecation message so developers know this is coming.

```
arm-mingw-dynamic.cmake
arm-mingw-static.cmake
arm-uwp-static-md.cmake
arm-uwp.cmake
arm-windows-static.cmake
arm-windows.cmake
```

**Reply by BillyONeal** [reply]
We don't currently have a great mechanism for triplets to say that they are deprecated; is there a similar warning emitted by MSBuild / the build tools / etc. whose wording we should match?

**Reply by walbourn** [reply]
The only messaging I've seen so far is this:

https://learn.microsoft.com/en-us/windows/arm/arm32-to-arm64

**arm32** desktop has never been officially, so it's just **arm-uwp** that they ever supported.

**Reply by BillyONeal** [reply]
OK, let me ask around for what the rest of the dev tools are doing on vcsig. (IMO the main place that needs to see a warning consistent with the above are MSBuild customers, as I expect most messages we might emit are likely to be dropped on the floor)
